### PR TITLE
Solution: bug #478

### PR DESCRIPTION
**Pull request by zymonmikolaj**
Fix params for Binance's API getTradeHistory()

### CODE REVIEW

File: spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java [hunk 1]
```diff
@@ -250,27 +254,37 @@ public final boolean cancelOrder(final String orderId) {
...
-            params.setEndTime(endDate);
+            params.setEndTime(now);
+
+            Set<TradeDTO> results = new LinkedHashSet<>();
 
             // Set currency pairs (required for exchanges like Gemini or Binance).
             if (!currencyPairs.isEmpty()) {
-                params.setCurrencyPairs(currencyPairs.stream().map(currencyMapper::mapToCurrencyPair).collect(Collectors.toCollection(LinkedList::new)));
+
+                currencyPairs.forEach(pair -> {
```
Comment (straumat) [inline spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java:267]: Is it necessary to make a call by currency pair ? setCurrencyPairs doesn't work ?

File: spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java [hunk 2]
```diff
@@ -250,27 +254,37 @@ public final boolean cancelOrder(final String orderId) {
...
+
+                    params.setCurrencyPair(currencyMapper.mapToCurrencyPair(pair));
+
+                    try {
+                        results.addAll(
+                                tradeService.getTradeHistory(params)
```
Comment (straumat) [inline spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java:273]: If we have to make one call by currency pair, we will have to add getBucket().asScheduler().consume(1); to we won't be ban for calling too often the exchange

**Review by straumat — COMMENTED** [review]
Can you reply to my questions ? Thx

File: spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java (hunk 1 above)
Comment (zymonmikolaj) [reply spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/service/xchange/TradeServiceXChangeImplementation.java:267]: Yes, inside XChange CurrencyPairs is ignored. Binance implementation use singe CurrencyPair.

**Pull request merged**